### ISSUE TITLE
It looks not fine in dumped topo xml file

### ISSUE THREAD

**Issue by yanminjia** [NVIDIA/nccl#1182]
On an H100 server, as the attached dumped topo file, GPU 7 has only one nvlink sub node, which is much different from other GPUs. Additionally, the target id looks invalid. 

<img width="1261" alt="image" src="https://github.com/NVIDIA/nccl/assets/6766475/9d72f524-856f-43de-af71-1b4bf43ca78c">

complete dumped topo file - 
[topo_dump.zip](https://github.com/NVIDIA/nccl/files/14328743/topo_dump.zip)

Basically, I think each GPU node should have 4 nvlink sub nodes because H100 has 4 NVSwitches.

**Comment by sjeaugey** [comment]
In some cases, like VMs, we may not be able to see the NVSwitches. So when we see NVLink but NVML can't identify the remote side, we have to assume those are invisible NVSwitches. That's why we set the target as class bridge and ID -1.

Now it's a bit weird that only one GPU is in that situation. They should all be the same.

Can you run `nvidia-smi nvlink -R` and see what it reports?

**Comment by yanminjia** [comment]
Many thanks for your prompt response @sjeaugey. Yes, as you said, GPU 7 failed to identify the remote target ID either in docker or in host system. In this case, I'm wondering if it would have impact on the performace of RedcueScatter or AllReduce. 

root@10-51-1-1:/workspace# nvidia-smi nvlink -R
GPU 0: NVIDIA H100 80GB HBM3 (UUID: GPU-257d8671-d0e6-d1a7-dea9-952249a668fa)
         Link 0: Remote Device 00000000:07:00.0: Link 46
         Link 1: Remote Device 00000000:07:00.0: Link 47
         Link 2: Remote Device 00000000:06:00.0: Link 2
         Link 3: Remote Device 00000000:06:00.0: Link 3
         Link 4: Remote Device 00000000:07:00.0: Link 41
         Link 5: Remote Device 00000000:07:00.0: Link 40
         Link 6: Remote Device 00000000:06:00.0: Link 5
         Link 7: Remote Device 00000000:06:00.0: Link 4
         Link 8: Remote Device 00000000:05:00.0: Link 46
         Link 9: Remote Device 00000000:05:00.0: Link 47
         Link 10: Remote Device 00000000:07:00.0: Link 34
         Link 11: Remote Device 00000000:06:00.0: Link 32
         Link 12: Remote Device 00000000:08:00.0: Link 34
         Link 13: Remote Device 00000000:08:00.0: Link 35
         Link 14: Remote Device 00000000:05:00.0: Link 43
         Link 15: Remote Device 00000000:05:00.0: Link 42
         Link 16: Remote Device 00000000:08:00.0: Link 38
         Link 17: Remote Device 00000000:08:00.0: Link 39
GPU 1: NVIDIA H100 80GB HBM3 (UUID: GPU-cc7d93b3-73f8-5028-287a-9e617839550e)
         Link 0: Remote Device 00000000:08:00.0: Link 47
         Link 1: Remote Device 00000000:08:00.0: Link 46
         Link 2: Remote Device 00000000:06:00.0: Link 50
         Link 3: Remote Device 00000000:06:00.0: Link 51
         Link 4: Remote Device 00000000:08:00.0: Link 43
         Link 5: Remote Device 00000000:08:00.0: Link 42
         Link 6: Remote Device 00000000:07:00.0: Link 3
         Link 7: Remote Device 00000000:07:00.0: Link 2
         Link 8: Remote Device 00000000:05:00.0: Link 33
         Link 9: Remote Device 00000000:05:00.0: Link 32
         Link 10: Remote Device 00000000:06:00.0: Link 53
         Link 11: Remote Device 00000000:07:00.0: Link 35
         Link 12: Remote Device 00000000:05:00.0: Link 37
         Link 13: Remote Device 00000000:05:00.0: Link 36
         Link 14: Remote Device 00000000:06:00.0: Link 62
         Link 15: Remote Device 00000000:06:00.0: Link 63
         Link 16: Remote Device 00000000:07:00.0: Link 38
         Link 17: Remote Device 00000000:07:00.0: Link 39
GPU 2: NVIDIA H100 80GB HBM3 (UUID: GPU-3af86863-b2c9-581e-fdbf-0935a4e104ec)
         Link 0: Remote Device 00000000:06:00.0: Link 36
         Link 1: Remote Device 00000000:06:00.0: Link 37
         Link 2: Remote Device 00000000:05:00.0: Link 40
         Link 3: Remote Device 00000000:05:00.0: Link 41
         Link 4: Remote Device 00000000:08:00.0: Link 58
         Link 5: Remote Device 00000000:08:00.0: Link 59
         Link 6: Remote Device 00000000:07:00.0: Link 62
         Link 7: Remote Device 00000000:07:00.0: Link 63
         Link 8: Remote Device 00000000:08:00.0: Link 63
         Link 9: Remote Device 00000000:08:00.0: Link 62
         Link 10: Remote Device 00000000:07:00.0: Link 45
         Link 11: Remote Device 00000000:06:00.0: Link 40
         Link 12: Remote Device 00000000:05:00.0: Link 44
         Link 13: Remote Device 00000000:05:00.0: Link 45
         Link 14: Remote Device 00000000:07:00.0: Link 43
         Link 15: Remote Device 00000000:07:00.0: Link 42
         Link 16: Remote Device 00000000:06:00.0: Link 46
         Link 17: Remote Device 00000000:06:00.0: Link 47
GPU 3: NVIDIA H100 80GB HBM3 (UUID: GPU-67d76fa4-2a89-ca9a-2071-ad52f7c12dab)
         Link 0: Remote Device 00000000:08:00.0: Link 61
         Link 1: Remote Device 00000000:08:00.0: Link 60
         Link 2: Remote Device 00000000:06:00.0: Link 39
         Link 3: Remote Device 00000000:06:00.0: Link 38
         Link 4: Remote Device 00000000:08:00.0: Link 57
         Link 5: Remote Device 00000000:08:00.0: Link 56
         Link 6: Remote Device 00000000:05:00.0: Link 53
         Link 7: Remote Device 00000000:05:00.0: Link 52
         Link 8: Remote Device 00000000:07:00.0: Link 50
         Link 9: Remote Device 00000000:07:00.0: Link 51
         Link 10: Remote Device 00000000:06:00.0: Link 33
         Link 11: Remote Device 00000000:07:00.0: Link 52
         Link 12: Remote Device 00000000:05:00.0: Link 49
         Link 13: Remote Device 00000000:05:00.0: Link 48
         Link 14: Remote Device 00000000:07:00.0: Link 16
         Link 15: Remote Device 00000000:07:00.0: Link 17
         Link 16: Remote Device 00000000:06:00.0: Link 1
         Link 17: Remote Device 00000000:06:00.0: Link 0
GPU 4: NVIDIA H100 80GB HBM3 (UUID: GPU-9f3b2427-df1c-413a-cdc1-bdee18d97d9b)
         Link 0: Remote Device 00000000:07:00.0: Link 33
         Link 1: Remote Device 00000000:07:00.0: Link 32
         Link 2: Remote Device 00000000:08:00.0: Link 37
         Link 3: Remote Device 00000000:08:00.0: Link 36
         Link 4: Remote Device 00000000:07:00.0: Link 0
         Link 5: Remote Device 00000000:07:00.0: Link 1
         Link 6: Remote Device 00000000:05:00.0: Link 34
         Link 7: Remote Device 00000000:05:00.0: Link 35
         Link 8: Remote Device 00000000:06:00.0: Link 6
         Link 9: Remote Device 00000000:06:00.0: Link 7
         Link 10: Remote Device 00000000:07:00.0: Link 19
         Link 11: Remote Device 00000000:06:00.0: Link 42
         Link 12: Remote Device 00000000:08:00.0: Link 33
         Link 13: Remote Device 00000000:08:00.0: Link 32
         Link 14: Remote Device 00000000:05:00.0: Link 39
         Link 15: Remote Device 00000000:05:00.0: Link 38
         Link 16: Remote Device 00000000:06:00.0: Link 35
         Link 17: Remote Device 00000000:06:00.0: Link 34
GPU 5: NVIDIA H100 80GB HBM3 (UUID: GPU-983c5e86-21e9-19ac-b01c-93ce4e6e3956)
         Link 0: Remote Device 00000000:06:00.0: Link 60
         Link 1: Remote Device 00000000:06:00.0: Link 61
         Link 2: Remote Device 00000000:08:00.0: Link 50
         Link 3: Remote Device 00000000:08:00.0: Link 51
         Link 4: Remote Device 00000000:06:00.0: Link 45
         Link 5: Remote Device 00000000:06:00.0: Link 44
         Link 6: Remote Device 00000000:07:00.0: Link 59
         Link 7: Remote Device 00000000:07:00.0: Link 58
         Link 8: Remote Device 00000000:08:00.0: Link 54
         Link 9: Remote Device 00000000:08:00.0: Link 55
         Link 10: Remote Device 00000000:06:00.0: Link 41
         Link 11: Remote Device 00000000:07:00.0: Link 18
         Link 12: Remote Device 00000000:05:00.0: Link 56
         Link 13: Remote Device 00000000:05:00.0: Link 57
         Link 14: Remote Device 00000000:07:00.0: Link 54
         Link 15: Remote Device 00000000:07:00.0: Link 55
         Link 16: Remote Device 00000000:05:00.0: Link 61
         Link 17: Remote Device 00000000:05:00.0: Link 60
GPU 6: NVIDIA H100 80GB HBM3 (UUID: GPU-ad536355-482f-a87c-b081-8523fb3971c4)
         Link 0: Remote Device 00000000:06:00.0: Link 57
         Link 1: Remote Device 00000000:06:00.0: Link 56
         Link 2: Remote Device 00000000:07:00.0: Link 60
         Link 3: Remote Device 00000000:07:00.0: Link 61
         Link 4: Remote Device 00000000:08:00.0: Link 48
         Link 5: Remote Device 00000000:08:00.0: Link 49
         Link 6: Remote Device 00000000:05:00.0: Link 59
         Link 7: Remote Device 00000000:05:00.0: Link 58
         Link 8: Remote Device 00000000:08:00.0: Link 53
         Link 9: Remote Device 00000000:08:00.0: Link 52
         Link 10: Remote Device 00000000:07:00.0: Link 44
         Link 11: Remote Device 00000000:06:00.0: Link 52
         Link 12: Remote Device 00000000:05:00.0: Link 62
         Link 13: Remote Device 00000000:05:00.0: Link 63
         Link 14: Remote Device 00000000:07:00.0: Link 37
         Link 15: Remote Device 00000000:07:00.0: Link 36
         Link 16: Remote Device 00000000:06:00.0: Link 49
         Link 17: Remote Device 00000000:06:00.0: Link 48
GPU 7: NVIDIA H100 80GB HBM3 (UUID: GPU-82cbc906-5e93-ce56-51b9-5ef3ace78e75)
         Link 0: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 1: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 2: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 3: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 4: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 5: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 6: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 7: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 8: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 9: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 10: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 11: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 12: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 13: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 14: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 15: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 16: Remote Device FFFFFFFF:FF:FF.0: Link 0
         Link 17: Remote Device FFFFFFFF:FF:FF.0: Link 0
